### ISSUE TITLE
move rhel7 worker node updates entirely to openshift-ansible

### ISSUE THREAD

**Issue by cgwalters** [openshift/machine-config-operator#1592]
It's really confusing today the role and split between [openshift-ansible](https://github.com/openshift/openshift-ansible/) and the MCO (we need a FAQ for this).

Anyways here's an idea: what if we stopped the MCD running on !CoreOS systems, and instead openshift-ansible always needed to be run to apply changes from MachineConfig.  It'd be openshift-ansible that'd somehow trigger drain/reboot etc.

The downside of this is e.g. if a user wants to change the kubelet config, they `oc edit` but then need to bounce out of Kube down to Ansible+ssh.

But the massive benefit is that it's super clear how things work - the MCO can optimize for CoreOS, openshift-ansible optimizes fully for traditional.  

And with PRs like https://github.com/openshift/machine-config-operator/pull/1586/ for example - we can *rely* on `/usr/etc` for original files, and do things like [make better use of OSTree](https://github.com/openshift/machine-config-operator/issues/119).

**Comment by jlebon** [comment]
A random idea I discussed with @runcom which is less radical but more cumbersome is we detect the case where the cluster is only RHCOS nodes and activating "OSTree-optimized" mode where we can use things like `/usr/etc`. But yeah, not fun to maintain multiple paths to do the same thing.

**Comment by kikisdeliveryservice** [comment]
I like the clear "separation of powers" as I think it would allow us to optimize and maintain a tight and consistent focus - would also seem to help from a troubleshooting (fixing/docs/ux) perspective going forward.

How fussy this would be to implement: ¯\_(ツ)_/¯¯

**Comment by sdodson** [comment]
> Anyways here's an idea: what if we stopped the MCD running on !CoreOS systems, and instead openshift-ansible always needed to be run to apply changes from MachineConfig. It'd be openshift-ansible that'd somehow trigger drain/reboot etc.

Do you specifically mean running MCD as a daemon or would even the `once-from` mode be eliminated outside of CoreOS?

/cc @crawford

**Comment by cgwalters** [comment]
> Do you specifically mean running MCD as a daemon or would even the once-from mode be eliminated outside of CoreOS?

Just the MCD as daemon.

**Comment by cgwalters** [comment]
One thing I could imagine is that openshift-ansible owns a "translator" from MachineConfig into Ansible...e.g. if `kernelArguments` in a MC are set that could be a playbook that calls out to `grubby`.

**Comment by cgwalters** [comment]
To elaborate on this, today openshift-ansible [applies machineconfig updates](https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_node/tasks/upgrade.yml#L32) - in doing this we'd be making openshift-ansible work the same way the MCO does: we don't distinguish between "configuration change" and "upgrade".  And not making that distinction is the correct thing to do because it ensures there's only one code path which needs to work.

**Comment by cgwalters** [comment]
One other thing about this is that if we stop running the MCD on rhel7, then we can move the MCD container image to RHEL8, which would make it *significantly* saner to also extract the binary to the host and run it there.

**Comment by michaelgugino** [comment]
The suggestions here seem to be going in the complete opposite direction.  We should be making efforts to manage more within the cluster, not pushing more into openshift-ansible.

**Comment by cgwalters** [comment]
> The suggestions here seem to be going in the complete opposite direction. We should be making efforts to manage more within the cluster, not pushing more into openshift-ansible.

But people using BYO RHEL7 are presumably using it precisely because they have nontrivial investment in existing external management systems, non-containerized system software agents etc.  That just directly conflicts with moving things into the cluster and containerizing - the more you do that the more the question becomes "why aren't you using RHCOS"?

**Comment by michaelgugino** [comment]
> But people using BYO RHEL7 are presumably using it precisely because

I don't agree with your conclusions.  I think the primary reason to use RHEL7 is hardware/software/compliance that is only certified/works for/on RHEL7.  That's why we only support RHEL7 workers, to enable workloads that wouldn't other wise be possible.

IMO, there's no reason we can't support RHEL with MCO/MCD the same way we support RHCOS.
